app/javascript/components/cloud-object-store-container-form/index.jsx

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable camelcase */
+import React, { useState } from 'react';
+import { Loading } from 'carbon-components-react';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import createSchema from './cloud-object-store-container-form.schema';
+import { API } from '../../http_api';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const CloudObjectStoreContainerForm = () => {
+  const [{
+    initialValues, fields, isLoading,
+  }, setState] = useState({});
+
+  const loadSchema = (appendState = {}) => ({ data: { form_schema: { fields } } }) => {
+    miqSparkleOff();
+    setState({
+      ...appendState,
+      fields,
+    });
+  };
+
+  const emptySchema = (appendState = {}) => {
+    const fields = [];
+    setState((state) => ({
+      ...state,
+      ...appendState,
+      fields,
+    }));
+  };
+
+  const onSubmit = (values) => {
+    // AWS values match old form, need to ask about Openstack values
+    const submitData = {
+      name: values.name,
+      ems_id: values.storage_manager_id,
+    };
+    const url = '/cloud_object_store_container/show_list#/';
+
+    if (values.provider_region) {
+      submitData.region = values.provider_region;
+    } else if (values.cloud_tenant_id) {
+      submitData.cloud_tenant_id = values.cloud_tenant_id;
+    }
+    miqSparkleOn();
+    API.get(`/api/providers/${values.storage_manager_id}?attributes=type,parent_manager.type`)
+      .then((data) => {
+        submitData.emsType = data.type;
+        submitData.parent_emstype = data.parent_manager.type;
+      })
+      .then(() => {
+        API.post('/api/cloud_object_store_containers/', submitData).then(() => {
+          const message = __(`Add of Cloud Object Store Container ${values.name} has been successfully queued.`);
+          miqRedirectBack(message, 'success', url);
+        })
+          .catch((error) => {
+            const message = __(`Add of Cloud Object Store Container ${values.name} has failed with error: ${error.data.error.message}`);
+            miqRedirectBack(message, 'warning', url);
+          });
+      });
+  };
+
+  const onCancel = () => {
+    miqSparkleOn();
+    const url = '/cloud_object_store_container/show_list#/';
+    const message = 'Adding of new cloud object store container was cancelled by the user.';
+    miqRedirectBack(message, 'warning', url);
+  };
+
+  const validation = (values) => {
+    const errors = {};
+    if (values.storage_manager_id === '-1') {
+      errors.storage_manager_id = __('Required');
```

Comment:
Why do we needs this ? Can't we use  ` isRequired: true` and   `validate: [{ type: validatorTypes.REQUIRED }]`?

Reply:
This is required due to the manual placeholder. The manual placeholder is used for emptyschema and this function disables the submit button when the schema is cleared.